samples/csharp/getting-started/AdvancedExperiment_AutoML/README.md

```diff
@@ -0,0 +1,141 @@
+    new[] { new KeyValuePair<string, bool>("CSH", true) }, "payment_type");
+```
+
+## Step 4: Edit Inferred Column Information
+Edit or correct the column information that was inferred:
+
+```C#
+ColumnInformation columnInformation = columnInference.ColumnInformation;
+columnInformation.CategoricalColumnNames.Remove("payment_type");
+columnInformation.IgnoredColumnNames.Add("payment_type");
+```
+
+## Step 5: Initialize Experiment Settings
+
+Initialize the AutoML experiment settings:
+
+```C#
+var experimentSettings = new RegressionExperimentSettings();
+experimentSettings.MaxExperimentTimeInSeconds = 3600;
+
+// Set the metric that AutoML will try to optimize over the course of the experiment.
+experimentSettings.OptimizingMetric = RegressionMetric.RootMeanSquaredError;
+
+// Set the cache directory to null.
+// This will cause all models produced by AutoML to be kept in memory 
+// instead of written to disk after each run, as AutoML is training.
+experimentSettings.CacheDirectory = null;
+
+// Don't use LbfgsPoissonRegression and OnlineGradientDescent trainers during this experiment.
+experimentSettings.Trainers.Remove(RegressionTrainer.LbfgsPoissonRegression);
+experimentSettings.Trainers.Remove(RegressionTrainer.OnlineGradientDescent);
+
+// Cancel experiment after the user presses any key
+experimentSettings.CancellationToken = cts.Token;
+CancelExperimentAfterAnyKeyPress(cts);
+```
+
+## Step 6: Create the AutoML Experiment
+
+Create the AutoML experiment using the initialized experiment settings:
+
+```C#
+var experiment = mlContext.Auto().CreateRegressionExperiment(experimentSettings);
+```
+
+## Step 7: Execute the AutoML Experiment
+
+Executing triggers data featurization, learning algorithm selection, hyperparameter tuning, etc. AutoML will iterate over different data featurizations, machine learning algorithms, hyperparamters, etc. until `MaxExperimentTimeInSeconds` is reached or the experiment is terminated.
+
+```C#            
+ ExperimentResult<RegressionMetrics> experimentResult = experiment.Execute(TrainSmallDataView, columnInformation, preFeaturizer, progressHandler);
+```
+
+## Step 8: Re-fit Best Pipeline
+Re-fit the best pipeline (trained from subsample of AutoML data) on entirety of training data. (This step is optional. By no means is it required in your workflow.)
+
+```C#
+private static ITransformer RefitBestPipeline(MLContext mlContext, ExperimentResult<RegressionMetrics> experimentResult)
+{
+    RunDetail<RegressionMetrics> best = experimentResult.BestRun;
+    return best.Estimator.Fit(TrainDataView);
+}
+```
+
+## Step 9: Evaluate Model
+
+Evaluate the quality of the re-fit model on a test dataset that was not used in training (`taxi-fare-test.csv`).
+
+```C#
+IDataView predictions = model.Transform(TestDataView);
+var metrics = mlContext.Regression.Evaluate(predictions, labelColumnName: LabelColumnName, scoreColumnName: "Score");
+```
+
+## Step 10: Make Predictions
```

Comment:
There should be a refit on the entire dataset (Train+Test). Metrics are produced in step 9, the actual model to send to production is created in this final refit.

Reply:
Moved re-fit for to the end in this advanced sample that re-fits on train & test, and then saves that model for use in production